viadot/sources/genesys.py

```diff
@@ -405,6 +346,74 @@ def download_all_reporting_exports(
             self.logger.info("Successfully genetared file names list.")
             return file_name_list
 
+    def generate_reporting_export(
+        self, data_to_post: Dict[str, Any], verbose: bool = False
+    ) -> int:
+        """
+        POST method for reporting export.
+
+        Args:
+            data_to_post (Dict[str, Any]): json format of POST body.
+            verbose (bool, optional): Decide if enable logging.
+
+        Returns:
+            new_report.status_code: status code
+        """
+        payload = json.dumps(data_to_post)
+        new_report = handle_api_response(
+            url=f"https://api.{self.environment}/api/v2/analytics/reporting/exports",
+            headers=self.authorization_token,
+            method="POST",
+            body=payload,
+        )
+        if verbose:
+            if new_report.status_code == 200:
+                self.logger.info("Succesfully generated new export.")
+            else:
+                self.logger.error(
+                    f"Failed to generated new export. - {new_report.content}"
+                )
+                raise APIError("Failed to generated new export.")
+        return new_report.status_code
+
+    def delete_reporting_exports(self, report_id):
+        """DELETE method for deleting particular reporting exports.
+
+        Args:
+            report_id (str): defined at the end of report url.
+
+        Returns:
+            delete_method.status_code: status code
+        """
+        delete_method = handle_api_response(
+            url=f"https://api.{self.environment}/api/v2/analytics/reporting/exports/{report_id}",
+            headers=self.authorization_token,
+            method="DELETE",
+        )
+        if delete_method.status_code < 300:
+            self.logger.info("Successfully deleted report from Genesys API.")
+
+        else:
+            self.logger.error(
+                f"Failed to deleted report from Genesys API. - {delete_method.content}"
+            )
+            raise APIError("Failed to deleted report from Genesys API.")
+
+        return delete_method.status_code
+
+    def delete_all_reporting_exports(self):
+        """
+        Function that deletes all reporting from self.reporting_data list.
+
+        Returns:
```

Comment:
You have a `returns` section, but the function definition does not provide the return type. In function, you don't have `return`. Is it correct?

Reply:
Another legacy.
Removed from docs and added `-> None` type return.